pkg/domain/workflow.go

```diff
@@ -9,17 +9,21 @@ import (
 
 // WorkflowStore is an inteface to workflow stores
 type WorkflowStore interface {
-	Find(ctx context.Context, name string) *workflow.Workflow
-	GetAll(ctx context.Context, name string) (result []*workflow.Workflow)
-	Add(ctx context.Context, r *workflow.Workflow) (*workflow.Workflow, error)
+	GetWorkflow(ctx context.Context, name string) *workflow.Workflow
+	GetAllWorkflows(ctx context.Context, name *string) (result []*workflow.Workflow)
+	AddWorkflow(ctx context.Context, w *workflow.Workflow) (*workflow.Workflow, error)
+	GetAssignedCodesets(ctx context.Context, workflowName string) []*Codeset
+	GetAssignments(ctx context.Context, workflowName *string) map[string][]*Codeset
+	AddCodesetAssignment(ctx context.Context, workflowName string, codeset Codeset) []*Codeset
```

Comment:
Not sure this is intended: you should avoid passing larger structs as values. Pass them as reference (pointer) instead. Keep in mind though, that the workflow store will keep a pointer to a codeset struct that is also in use by the codeset store (not that I find that to be a problem).

Reply:
thanks, I did not know about that